plugin/trino-open-policy-agent/src/test/java/io/trino/plugin/openpolicyagent/OpaAccessControlFilteringUnitTest.java

```diff
@@ -0,0 +1,333 @@
+                tableThree, Set.of("table-three-column-one", "table-three-column-two"));
+        // Allow both columns from one table, one column from another one and no columns from the last one
+        Set<String> columnsToAllow = Set.of("table-one-column-one", "table-one-column-two", "table-two-column-two");
+
+        this.mockClient.setHandler(buildHandler("/input/action/resource/table/columns/0", columnsToAllow));
+
+        Map<SchemaTableName, Set<String>> result = authorizer.filterColumns(
+                requestingSecurityContext,
+                "my-catalog",
+                requestedColumns);
+
+        List<String> expectedRequests = requestedColumns
+                .entrySet()
+                .stream()
+                .<String>mapMulti(
+                        (requestedColumnsForTable, accepter) -> requestedColumnsForTable.getValue().forEach(
+                                (col) -> accepter.accept("""
+                                    {
+                                        "operation": "FilterColumns",
+                                        "resource": {
+                                            "table": {
+                                                "tableName": "%s",
+                                                "schemaName": "my-schema",
+                                                "catalogName": "my-catalog",
+                                                "columns": ["%s"]
+                                            }
+                                        }
+                                    }""".formatted(requestedColumnsForTable.getKey().getTableName(), col))))
+                .collect(Collectors.toList());
+        assertStringRequestsEqual(expectedRequests, this.mockClient.getRequests(), "/input/action");
+        assertTrue(
+                Maps.difference(
+                        result,
+                        Map.of(
+                                tableOne, Set.of("table-one-column-one", "table-one-column-two"),
+                                tableTwo, Set.of("table-two-column-two"),
+                                tableThree, Set.of()))
+                        .areEqual());
+    }
+
+    @Test
+    public void testEmptyFilterColumns()
+    {
+        SchemaTableName someTable = SchemaTableName.schemaTableName("my-schema", "my-table");
+        Map<SchemaTableName, Set<String>> requestedColumns = Map.of(someTable, Set.of());
+
+        Map<SchemaTableName, Set<String>> result = authorizer.filterColumns(
+                requestingSecurityContext,
+                "my-catalog",
+                requestedColumns);
+
+        assertEquals(mockClient.getRequests().size(), 0);
+        assertTrue(
+                Maps.difference(
+                        result,
+                        requestedColumns).areEqual());
+    }
+
+    @ParameterizedTest(name = "{index}: {0}")
+    @MethodSource("io.trino.plugin.openpolicyagent.FilteringTestHelpers#emptyInputTestCases")
```

Comment:
Got it. Should I rewrite all of them or can we leave them as is for now?

Since we attempt to cover all possible corner cases, there is quite a bit of parameterization at the moment indeed

Reply:
To clarify, we are moving away from usages of annotations like `@ParameterizedTest` as they are hard to follow and debug.  Lots of test abuse these system to do things like call a test with a boolean to test with and without nulls, when it is way clearer to future readers to just see two tests or a test with two invocations (with and without nulls).  Additionally there are tests that have massive providers (hundreds of cases) and then it is impossible to fix just one of the cases when something changes.  In all of these cases it easier to understand and maintain with explicit cases in the code (not interpolated into the code).

That said there are _rare_ places where this kind of style makes sense, and in those places we recommend that you use a for loop over the data (possibly add an enum for the data).

Reply:
> To clarify, we are moving away from usages of annotations like `@ParameterizedTest` as they are hard to follow and debug.

(For the record, not everyone agrees with this assessment. But @dain has the final word on it.)